### DISCUSSION TITLE
Is the Platform Runtime Binary download supposed to contain source bundles?

### DISCUSSION THREAD

**Discussion by nitind** [eclipse-platform/eclipse.platform#1491] [General]
I was looking at the using the 4.32 platform download in a CI process, and puzzled at its size. R4.31 and R4.32 both include a number of source bundles that R4.30 and earlier didn't. 4.30's equivalent download is 97MB, not a difficult to understand increase from 4.8's 81MB, and neither had source bundles in them. R4.32's linux-gtk-x86_64 platform runtime download is 145MB, 50% 
 (and nearly 50MB) larger than R4.30.

This isn't the SDK download, and the source features don't seem to be present, so was this intentional or an oversight?

**Comment by nitind** [comment]
@akurtakov ?

**Reply by akurtakov** [reply]
This doesn't seem right to me. If you/someone can investigate what/why/how includes source bundles that would be very good first step towards fixing it.

**Comment by iloveeclipse** [comment]
I guess it is oversight. I remember I've opened a ticket to include source bundles by default in SDK, as it was hard to debug 3rd party code without sources. I guess the result is what you see now.

If you believe runtime shouldn't include sources (makes sense), may be you could try to fix that too?

**Reply by nitind** [reply]
Do you know what issue number it was? I can't find it.

**Comment by laeubi** [comment]
The SDK is for development purpose so I don't see why it should not contain sources and as mentioned this was even explicitly requested by @iloveeclipse so I don't see any problem/issue here, source features an no longer required and we better get rid of them sooner than later anyways.

Most build setups would use the updatesites anyways and putting effort into optimizing downloadsize it not a primary goal of platform sdk.

**Reply by nitind** [reply]
You seem to be addressing a different download and scenario than I am.

**Reply by laeubi** [reply]
The reason might be your description is quite vague, so that's totally possible. You are mentioning that

> I was looking at the using the 4.32 platform download in a CI process

- So what exact download URL you are using?!?
- What CI process and why does size matters at all (I assume you only download it once to a shared location e.g. nexus and then reuse it within your corporate network)
- Why using a Zip of unspecified content instead of e.g create your own if size/stability of content is a major concern for you?
- ...

**Reply by mickaelistria** [reply]
On the downlaod page such as https://download.eclipse.org/eclipse/downloads/drops4/R-4.32-202406010610/ , there is a "Platform Runtime binary" which is exactly what's referred in the title.
I agree with @nitind here that having the sources here seems unexpected, as it's really a "runtime binary", and sources are neither runtime, nor binary. The goal of this artifact is to get a minimal artifact that provides most Platform features, this is typically suitable for development of RCP applications that are not IDEs.

**Reply by merks** [reply]
Specifically these are described as binaries and so one would not expect source:

![image](https://github.com/user-attachments/assets/c78ca6fa-7105-4244-8853-c562598258a3)
